### PR TITLE
feat: bump node core to 5.0.0

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit bumps the node core dependency to 5.0.0, and also bumps the min supported version of node to be 18.

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [ ] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [ ] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,12 +36,12 @@
     "access": "public"
   },
   "engines": {
-    "node": ">=16.0.0"
+    "node": ">=18.0.0"
```
Comment (dpopp07) [inline package.json:39]: I'm not sure I really thought this through when I updated this value in the Node Core. _Technically_, this is a breaking change. I didn't necessarily want existing SDKs to have to release new major versions just to use the major version of the core.

Is it possible to install the new version of the core without making this change?

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

File: package.json (hunk 1 above)
Comment (padamstx) [reply package.json:39]: discussed offline

**Pull request merged**